ModularTegustation/tegu_items/rcorp/button.dm

```diff
@@ -0,0 +1,34 @@
+/obj/machinery/button/door/indestructible/rcorp
+	name = "rcorp button"
+	id = "inside"
+
+/obj/machinery/button/door/indestructible/rcorp/setup_device()
+	if(!device)
+		var/obj/item/assembly/control/rcorp/A = new(src)
+		device = A
+	..()
+
+/obj/machinery/button/door/indestructible/rcorp/Destroy()
+	qdel(device)
+	. = ..()
```

Comment:
Generally, Destroy() should `return ..()` and not `. = ..()`, prevents some copy-paste mistakes
```suggestion
	return ..()
```